drivers/flash/spi_flash_sfdp.c

```diff
@@ -172,12 +180,16 @@ static inline int spi_flash_write_sr_cr(struct device *dev, u8_t sr, u8_t cr)
 		.buf = buf,
 		.len = sizeof(buf),
 	};
+	struct spi_buf_set cmd_set = {
+		.buffers = &cmd,
+		.count = 1	
+	};
 
 	buf[0] = CMD_WRITE_STATUS_CONFIG;
 	buf[1] = sr;
 	buf[2] = cr;
 
-	if (spi_transceive(&data->config, &cmd, 1, &cmd, 1) != 0) {
+	if (spi_transceive(data->spi, &data->config, &cmd_set, &cmd_set) != 0) {
```

Comment:
looks like spi_write() would be just fine here.

Reply:
```
static inline int spi_write(struct device *dev,
			    const struct spi_config *config,
			    const struct spi_buf_set *tx_bufs)
{
	return spi_transceive(dev, config, tx_bufs, NULL);
}
```

Reply:
Sure, but as you can see this transaction does not care about what was read. So you can ditch the rx_bufs parameter, thus you can use spi_write directly.